[MINOR] fix: Align LogContext setup for core components

jeqo opened the pull request:
By not having a space after closing the custom prefix, logs end up in the wrong format and potentially break log parsing tools.

kamalcph reviewed (approved):
LGTM

divijvaidya commented:
Can we add a logic in LogContext constructor itself to add a suffic space if not already present? It would help in future cases where someone forgets to add a space.

jeqo commented:
@divijvaidya good call. Haven't looked into other components (connect, streams) to validate if they are using a different pattern. Would it be ok to do that in a separate PR?

jeqo committed:
feat: adding LogContext builder

Currently, log context is initialized with a custom log prefix. Though the format is hard to keep aligned. Adding a builder to ensure consistency.

jeqo committed:
refactor: use log context builder instead of hard-coded prefix

jeqo commented:
@divijvaidya I took a stab on refactoring the LogContext by adding a builder and adopting it on most of the source code but stream module as there is a heavy dependency on logPrefix as a string on APIs, so would require further refactoring than just replacing with a builder.

jeqo committed:
fixup! refactor: use log context builder instead of hard-coded prefix

Fallback changes on ShutdownableThread and Logging to avoid collision

jeqo committed:
fixup! refactor: use log context builder instead of hard-coded prefix

Refactor: rename builder for easier reading.
